src/main/java/com/jfrog/ide/common/nodes/VulnerabilityNode.java

```diff
@@ -121,25 +119,34 @@ public ResearchInfo getResearchInfo() {
         return researchInfo;
     }
 
-    public List<ApplicableIssueNode> getApplicableIssues() {
-        return applicableIssues;
+    public ApplicableInfo getApplicableInfo() {
+        return applicableInfo;
     }
 
-    public void setApplicableIssues(List<ApplicableIssueNode> applicableIssues) {
-        this.applicableIssues = applicableIssues;
+    /**
+     * If no previous applicability evidences exist, set the applicable info as "no applicable".
+     */
+    public void setNotApplicable() {
+        if (this.applicableInfo == null) {
+            this.applicableInfo = new ApplicableInfo(false);
+        }
     }
 
-    public void addApplicableIssue(ApplicableIssueNode issue) {
-        this.applicableIssues = this.applicableIssues == null ? new ArrayList<>() : this.applicableIssues;
-        this.applicableIssues.add(issue);
+
+    public void updateApplicableInfo(ApplicableIssueNode issue) {
+        if (this.applicableInfo == null) {
+            this.applicableInfo = new ApplicableInfo(true, issue.getScannerSearchTarget(), issue.getReason(), issue.getFilePath(), issue.getLineSnippet());
+        } else {
+            applicableInfo.addInfo(issue.getReason(), issue.getFilePath(), issue.getLineSnippet());
+        }
     }
 
     /**
      * Returns true if the issue is applicable, false if not.
      * If the applicability status is unknown, null will be returned.
      */
     public Boolean isApplicable() {
```

Comment:
If isApplicable is a boolean, its default value is already false. Is there any extra value in a null return value?
```suggestion
    public boolean isApplicable() {
```

Reply:
Yes. There are 3 stats - Applicable, Not Applicable, and Unkown